src/EchoParameterDynamicFunctionReturnTypeExtension.php

```diff
@@ -78,13 +78,44 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 
         if ($echoArgumentType instanceof ConstantBooleanType) {
             return ($echoArgumentType->getValue() === false)
-                ? new StringType()
-                : new VoidType();
+                ? self::getEchoFalseReturnType($name)
+                : self::getEchoTrueReturnType($name);
         }
 
         return TypeCombinator::union(
-            new StringType(),
-            new VoidType()
+            self::getEchoFalseReturnType($name),
+            self::getEchoTrueReturnType($name)
         );
     }
+
+    protected static function getEchoTrueReturnType( string $name ): Type
```

Comment:
Why are these methods static?

Reply:
Because they don't need an instantiation of the class

Reply:
Okay!